number_search.py

```diff
@@ -3,25 +3,46 @@
 # use regex to match numbers and fractions
 import re
 
+
 def count_numbers(text):
     """
         counts whole number and decimals i.e -ve and +ve whole numbers decimals
         TODO update doc strings
     """
+    count = 0
+    pattern = re.compile(r'(\d+| \d+\.\d+)')
```

Comment:
good job on the pattern.. but it will capture anything number no matter what is preceding it i.e. 12years or 2x or 23:34 or 2.3.3 are not numbers but your pattern will capture it. so update the pattern